packages/slate/src/operations/apply.js

```diff
@@ -287,12 +287,20 @@ const APPLIERS = {
     const { anchorKey, focusKey, anchorOffset, focusOffset } = selection
     let node = document.assertPath(path)
 
-    if (anchorKey == node.key && anchorOffset >= rangeOffset) {
-      selection = selection.moveAnchor(-length)
+    if (anchorKey == node.key) {
```

Comment:
There now two of these if's, should be moved into one if:

```
    if (anchorKey == node.key) {
      if (anchorOffset >= rangeOffset) {
        selection = selection.moveAnchor(-length)
      } else if (anchorOffset > offset) {
        selection = selection.moveAnchorTo(anchorKey, offset)
      }
      if (focusOffset >= rangeOffset) {
        selection = selection.moveFocus(-length)
      } else if (focusOffset > offset) {
        selection = selection.moveFocusTo(focusKey, offset)
      }
    }
```

Reply:
Wouldn't that change the behavior? In that case, we'd only check anchorKey, not focusKey.

Reply:
No, just move them into one parent if. There are now two of the same if.

Reply:
Think I need glasses. Read them as the same if. Please ignore :)